Dockerfile.multistage

```diff
@@ -0,0 +1,28 @@
+# An example using multi-stage image builds
+
+# First, build the application in the `/app` directory
+FROM ghcr.io/astral-sh/uv:python3.12-bookworm-slim AS builder
+WORKDIR /app
+RUN --mount=type=cache,target=/root/.cache/uv \
+    --mount=type=bind,source=uv.lock,target=uv.lock \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    uv sync --frozen --no-install-project
+ADD . /app
+RUN --mount=type=cache,target=/root/.cache/uv \
+    uv sync --frozen
+
+
+# Then, use a final image without uv
+FROM python:3.12-slim-bookworm
+# It is important to use the image that matches the builder, as the path to the
+# Python executable must be the same, e.g., using `python:3.11-slim-bookworm`
+# will fail.
+
+# Copy the application from the builder
+COPY --from=builder --chown=app:app /app /app
+
+# Place executables in the environment at the front of the path
+ENV PATH="/app/.venv/bin:$PATH"
+
+# Run the FastAPI application by default
+CMD ["fastapi", "dev", "--host", "0.0.0.0", "/app/src/uv_docker_example"]
```

Comment:
Is it intentional that this uses `dev` instead of `run`?

Reply:
So far, these examples target the dev experience with bind mounts — so yeah. We should probably be clear about what you'd want different in production builds.